### ISSUE TITLE
Chart events not exposed - Feature request

### ISSUE THREAD

**Issue by temecom** [Glavin001/ember-c3#37]
As a C3 user/developer it would be nice to have the chart actions linked to the ember action chain. This would provide interaction with other components on the view eg: 

A group selection on the chart could trigger a refresh of a table showing the data details. 

This would involve catching the 'once' method callback in D3 and emitting it as an Ember action. 

Thanks

Chris (expatriate Canadian in San Diego)

**Comment by Glavin001** [comment]
Absolutely agree. This is planned. For whatever reason I neglected to create an Issue. Thank you!

**Comment by Glavin001** [comment]
I would love to receive a Pull Request adding this functionality :smile:.

**Comment by temecom** [comment]
I will give it a shot.

I looked for the D3 instance but could not find it. Where is it buried?

Chris

**Comment by Glavin001** [comment]
I think the applicable code would be https://github.com/Glavin001/ember-c3/blob/master/addon/components/c3-chart.js#L118-L120

**Comment by Glavin001** [comment]
How about this for expected usage:

```handlebars
{{c3-chart data=model chart-created=(action "chartCreated")}
```

And your Controller or Route would have an action named `chartCreated` that could save the chart instance passed to itself to be used later.

```javascript
{
    actions: {
        chartCreated: function(chart) {
            this.set('myChartProperty', chart);
        }
    }
}
```

This would also work without any internal changes to Ember-C3:

```handlebars
{{c3-chart data=model chart-created=(mut chart)}
```

Where `(mut chart)` would generate a function that access the chart instance and would set the property `chart` on the controller.

**Comment by maxwondercorn** [comment]
See issue #77

**Comment by maxwondercorn** [comment]
chart actions/events were available on the component but have been updated with pr #79.  Dummy app has examples using closure actions on chart events.  Update will be published when issue #77 checklist is completed
